### PR TITLE
Show width and height in inspector overlay

### LINKED ISSUE #6871: Show height and width dimensions on on device inspector overlay
The fix for this will actually be on the Framework side, but tracking here to add respective devtools labels and projects.

Screenshot 2023-11-30 at 3 22 11 PM

We already have this information from the Layout Explorer, so we can use that as a breadcrumb of where the height / width data is.

### PR DESCRIPTION
**Pull request by CoderDake**
Fixes https://github.com/flutter/devtools/issues/6871

Add the width and height to the inspector overlay. 1 decimal precision is used since that matches the way Devtools displays the values.

## Examples
Screenshot 2023-12-28 at 2 39 49 PM

Screenshot 2023-12-28 at 2 39 42 PM
Screenshot 2023-12-28 at 2 39 37 PM

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 1]
```diff
@@ -3188,11 +3188,15 @@ class _InspectorOverlayLayer extends Layer {
       }
       candidates.add(_TransformedRect(candidate, rootRenderObject));
     }
+    final _TransformedRect selectedRect = _TransformedRect(selected, rootRenderObject);
+    final String shortDescription = selection.currentElement!.toStringShort();
```
Comment (umairnawaz333) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:3192]:
```suggestion
    final String widgetName = selection.currentElement!.toStringShort();
```

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 2]
```diff
@@ -3188,11 +3188,15 @@ class _InspectorOverlayLayer extends Layer {
...
+    final String width = selectedRect.rect.width.toStringAsFixed(1);
+    final String height = selectedRect.rect.height.toStringAsFixed(1);
 
     final _InspectorOverlayRenderState state = _InspectorOverlayRenderState(
       overlayRect: overlayRect,
-      selected: _TransformedRect(selected, rootRenderObject),
-      tooltip: selection.currentElement!.toStringShort(),
+      selected: selectedRect,
+      tooltip: '$shortDescription($width x $height)',
```
Comment (umairnawaz333) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:3199]:
```suggestion
      tooltip: '$widgetName($width x $height)',
```

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 3]
```diff
@@ -3188,11 +3188,15 @@ class _InspectorOverlayLayer extends Layer {
...
+    final String widgetName = selection.currentElement!.toStringShort();
+    final String width = selectedRect.rect.width.toStringAsFixed(1);
+    final String height = selectedRect.rect.height.toStringAsFixed(1);
 
     final _InspectorOverlayRenderState state = _InspectorOverlayRenderState(
       overlayRect: overlayRect,
-      selected: _TransformedRect(selected, rootRenderObject),
-      tooltip: selection.currentElement!.toStringShort(),
+      selected: selectedRect,
+      tooltip: '$widgetName($width x $height)',
```
Comment (elliette) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:3199]: add space between widget name and (

**Comment by CoderDake** [comment]
I really messed up this branch trying to rebase it. I've re-opened the PR here on a clean branch: https://github.com/flutter/flutter/pull/140709

**Pull request closed without merging**